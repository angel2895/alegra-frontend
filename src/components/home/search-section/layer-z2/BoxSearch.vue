<template>
  <div class="search--box">
    <div class="search--box__icon-all">All</div>
    <input
      type="text"
      placeholder="Search Photo ..."
      v-model="word"
      @keypress="enterSearch"
    />
    <div class="search--box__icon" @click="searchWord">
      <img src="@/assets/icons/search.svg" alt="" />
    </div>
  </div>
</template>
<script lang="ts">
import { Options, Vue } from "vue-class-component";
import { mapActions } from "vuex";

import { GoogleImage } from "@/types/google-images";
import { EventKeyPress } from "@/types/event-keypress";

@Options({
  methods: mapActions(["getImagesGoogle"]),
})
export default class BoxSearch extends Vue {
  word = "";
  getImagesGoogle!: (word: string) => Promise<GoogleImage[]>;

  searchWord(): void {
    this.getImagesGoogle(this.word);
  }

  enterSearch(event: EventKeyPress): void {
    if (event.code === "Enter") {
      this.getImagesGoogle(this.word);
    }
  }
}
</script>

<style scoped>
.search--box {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.search--box__icon-all {
  display: none;
}

.search--box > input {
  width: 100%;

  padding: 10px;

  font-size: 1.4rem;

  border: 0;
  outline: none;

  border-radius: 10px 0px 0px 10px;
}

.search--box__icon {
  padding: 10px;

  align-self: center;

  background-color: rgb(211, 211, 211);
  border-radius: 0px 10px 10px 0px;
}

.search--box__icon > img {
  width: 20px;
}

.search--box__icon:active {
  background-color: rgb(192, 192, 192);
}

@media (min-width: 1300px) {
  .search--box > input {
    border-radius: 0px;
  }
  .search--box__icon-all {
    display: flex;
    align-items: center;
    padding: 0px 20px;

    font-size: 1.4rem;
    font-weight: bold;

    border-radius: 10px 0px 0px 10px;
    color: rgb(221, 220, 220);
    background-color: rgb(46, 46, 46);
  }
}
</style>
