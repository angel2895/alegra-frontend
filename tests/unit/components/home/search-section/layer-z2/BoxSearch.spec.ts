import { shallowMount } from "@vue/test-utils";
import { createStore } from "vuex";
// Components
import BoxSearch from "@/components/home/search-section/layer-z2/BoxSearch.vue";

const store = createStore({
  state() {
    return {};
  },
  actions: {
    getImagesGoogle() {},
  },
});

describe("BoxSearch.vue", () => {
  it("Match with snapshot", () => {
    const wrapper = shallowMount(BoxSearch, {
      global: {
        plugins: [store],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("The input has a properties for text input", () => {
    const wrapper = shallowMount(BoxSearch, {
      global: {
        plugins: [store],
      },
    });
    const $input = wrapper.find("input");
    expect($input.attributes("type")).toBe("text");
    expect($input.attributes("placeholder")).toBe("Search Photo ...");
  });

  it("Conexcion the input with v-model word", async (): Promise<void> => {
    const wrapper = shallowMount(BoxSearch, {
      global: {
        plugins: [store],
      },
    });
    await wrapper.find('input[type="text"]').setValue("apple");
    expect(wrapper.vm.word).toBe("apple");
  });

  it("Search when press icon search", async (): Promise<void> => {
    const getImagesGoogleSpy = jest.spyOn<any, any>(
      BoxSearch.methods,
      "getImagesGoogle"
    );
    const wrapper = shallowMount(BoxSearch, {
      global: {
        plugins: [store],
      },
    });
    const $button = wrapper.find(".search--box__icon");
    await $button.trigger("click");
    expect(getImagesGoogleSpy).toHaveBeenCalled();
  });

  it("Search the word when push the Enter key", async (): Promise<void> => {
    const getImagesGoogleSpy = jest.spyOn<any, any>(
      BoxSearch.methods,
      "getImagesGoogle"
    );
    const wrapper = shallowMount(BoxSearch, {
      global: {
        plugins: [store],
      },
    });
    const $input = wrapper.find("input");
    await $input.trigger("keypress", { code: "Enter" });
    expect(getImagesGoogleSpy).toHaveBeenCalled();
  });
});
