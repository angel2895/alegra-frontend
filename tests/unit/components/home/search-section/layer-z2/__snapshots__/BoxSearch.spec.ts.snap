// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxSearch.vue Match with snapshot 1`] = `
<div class="search--box">
  <div class="search--box__icon-all">All</div><input type="text" placeholder="Search Photo ...">
  <div class="search--box__icon"><img alt=""></div>
</div>
`;
